ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/connections/QUICConnection_v1.kt

```diff
@@ -726,6 +762,89 @@ internal class QUICConnection_v1(
         private val logger = logger()
     }
 
+    private inner class StreamManager {
+        private val receiveStates = hashMapOf<Long, ReceiveStreamState>()
+
+        private val streams = hashMapOf<Long, QUICStreamImpl>()
+
+        private val sendChannel: Channel<StreamFrame> = Channel(Channel.UNLIMITED)
+
+        init {
+            // todo close
+            CoroutineScope(Dispatchers.Default).launch {
+                while (isActive) {
+                    try {
+                        val frame = sendChannel.receive()
+                        sendInOneRTT(forceEndDatagram = frame.fin) { builder, hookConsumer ->
+                            writeStream(
+                                packetBuilder = builder,
+                                streamId = frame.streamId,
+                                offset = frame.offset,
+                                specifyLength = true,
+                                fin = frame.fin,
+                                data = frame.data
+                            )
+
+                            withAckFrameIfAny(builder, hookConsumer, EncryptionLevel.AppData)
+                        }
+                    } catch (e: Exception) {
+                        logger.error(e)
+                    }
+                }
+            }
+        }
+
+        suspend fun acceptStreamFrame(
+            streamId: Long,
+            offset: Long,
+            fin: Boolean,
+            streamData: ByteArray,
+        ) {
+            if (!receiveStates.containsKey(streamId)) {
+                val state = ReceiveStreamState(
+                    onDataReceived = {
+                        onDataReceived(streamId, it)
+                    },
+                    onClose = {
+                        receiveStates.remove(streamId)
+                    }
+                )
+
+                receiveStates[streamId] = state
+            }
+
+            receiveStates[streamId]!!.receive(streamData, offset, fin)
+        }
+
+        private suspend fun onDataReceived(streamId: Long, dataChunk: ByteReadPacket) {
+            if (!streams.containsKey(streamId)) {
```

Comment:
invert and early return

Reply:
There is a statement after `if` that should be executed, so I don't think it early return would improve the code here